lfs/hooks.go

```diff
@@ -0,0 +1,148 @@
+package lfs
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// A HookType represents a type of Git hook as defined in
+// http://git-scm.com/docs/githooks.
+type HookType string
+
+const (
+	ApplyPatchMessageHook    HookType = "applypatch-msg"
+	PreApplyPatchHook                 = "pre-applypatch"
+	PostApplyPatchHook                = "post-applypatch"
+	PreCommitHook                     = "pre-commit"
+	PrepareCommitMessageHook          = "prepare-commit-msg"
+	CommitMessageHook                 = "commit-msg"
+	PostCommitHook                    = "post-commit"
+	PreRebaseHook                     = "pre-rebase"
+	PostCheckoutHook                  = "post-checkout"
+	PostMergeHook                     = "post-merge"
+	PrePushHook                       = "pre-push"
+	PreReceiveHook                    = "pre-receive"
+	UpdateHook                        = "update"
+	PostReceiveHook                   = "post-receive"
+	PostUpdateHook                    = "post-update"
+	PushToCheckoutHook                = "push-to-checkout"
+	PostRewriteHook                   = "post-rewrite"
+	RebaseHook                        = "rebase"
```

Comment:
What value do we get from defining all of these hooks? You can turn any string into a `HookType` if you really want. I'd just make Hook's Type property a string and be done with it.

FWIW: this is similar to my own struggle with the `Config.Access()` [`AuthType` constants](https://github.com/github/git-lfs/blob/3226d2a652799ebf81026c1ed14d71b783239436/lfs/config.go#L17-L22). The only benefit the code gets is being able to compare against a constant instead of initializing the string. I'm not even sure that matters for a short lived command line tool.

So after being confronted with this again, I think I'm leaning the other way towards just using strings.